apps/emqx_ft/src/emqx_ft_assembler_sup.erl

```diff
@@ -0,0 +1,44 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2020-2023 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%--------------------------------------------------------------------
+
+-module(emqx_ft_assembler_sup).
+
+-export([start_link/1]).
+-export([start_child/3]).
+
+-behaviour(supervisor).
+-export([init/1]).
+
+-define(REF(ID), {via, gproc, {n, l, {?MODULE, ID}}}).
```

Comment:
For my own information: why do we use `gproc` to register the supervisor here?  It seems to be to allow using the tuple as the process alias, and something else?

Reply:
Honestly, we're not settled on what the IDs here are yet but I imagined that in the simplest case a directory name (a storage root) could be used as an `ID`, and `gproc` makes working with such identifiers a bit saner.